test/0.8.9/accounting.handleOracleReport.test.ts

```diff
@@ -1,652 +1,409 @@
+        oracleReportSanityChecker,
+        postTokenRebaseReceiver,
+        withdrawalQueue,
+        burner,
+      },
+      deployer,
+    );
+
+    const accountingImpl = await ethers.deployContract("Accounting", [locator, lido], deployer);
+    const accountingProxy = await ethers.deployContract(
+      "OssifiableProxy",
+      [accountingImpl, deployer, new Uint8Array()],
+      deployer,
+    );
+    accounting = await ethers.getContractAt("Accounting", accountingProxy, deployer);
+    await updateLidoLocatorImplementation(await locator.getAddress(), { accounting });
+    await accounting.initialize(deployer);
+
+    const accountingOracleSigner = await impersonate(await locator.accountingOracle(), ether("100.0"));
+    accounting = accounting.connect(accountingOracleSigner);
+  });
+
+  context("handleOracleReport", () => {
+    it("Update CL validators count if reported more", async () => {
+      const depositedValidators = 100n;
+      await lido.setMockedDepositedValidators(depositedValidators);
+
+      // second report, 101 validators
+      await accounting.handleOracleReport(
+        report({
+          clValidators: depositedValidators,
+        }),
+      );
+    });
+
+    it("Reverts if the `checkAccountingOracleReport` sanity check fails", async () => {
+      await oracleReportSanityChecker.mock__checkAccountingOracleReportReverts(true);
+
+      await expect(accounting.handleOracleReport(report())).to.be.reverted;
+    });
+
+    it("Reverts if the `checkWithdrawalQueueOracleReport` sanity check fails", async () => {
+      await oracleReportSanityChecker.mock__checkWithdrawalQueueOracleReportReverts(true);
+      await expect(
+        accounting.handleOracleReport(
+          report({
+            withdrawalFinalizationBatches: [1n],
+          }),
+        ),
+      ).to.be.reverted;
+    });
+
+    it("Does not revert if the `checkWithdrawalQueueOracleReport` sanity check fails but no withdrawal batches were reported", async () => {
+      await oracleReportSanityChecker.mock__checkWithdrawalQueueOracleReportReverts(true);
+      await withdrawalQueue.mock__isPaused(true);
+
+      await expect(accounting.handleOracleReport(report())).not.to.be.reverted;
+    });
+
+    /// NOTE: This test is not applicable to the current implementation (Accounting's _checkAccountingOracleReport() checks for checkWithdrawalQueueOracleReport()
+    /// explicitly in case _report.withdrawalFinalizationBatches.length > 0
+    // it("Does not revert if the `checkWithdrawalQueueOracleReport` sanity check fails but `withdrawalQueue` is paused", async () => {
+    //   await oracleReportSanityChecker.mock__checkWithdrawalQueueOracleReportReverts(true);
+    //   await withdrawalQueue.mock__isPaused(true);
+
+    //   await expect(accounting.handleOracleReport(report({ withdrawalFinalizationBatches: [1n] }))).not.to.be.reverted;
+    // });
+
+    it("Does not emit `StETHBurnRequested` if there are no shares to burn", async () => {
```

Comment:
Strange to check the actual event from Burner contract in Accounting unit tests. Probably, the mock event will be better.